test/sql/plan_hypertable_inline.sql

```diff
@@ -0,0 +1,24 @@
+-- This file and its contents are licensed under the Apache License 2.0.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-APACHE for a copy of the license.
+
+-- test hypertable classification when query is in an inlineable function
+
+\set PREFIX 'EXPLAIN (costs off)'
+
+CREATE TABLE test (a int, b bigint NOT NULL);
+SELECT create_hypertable('public.test', 'b', chunk_time_interval=>10);
+INSERT INTO test SELECT i, i FROM generate_series(1, 20) i;
+
+CREATE OR REPLACE FUNCTION test_f(_ts bigint)
+RETURNS SETOF test LANGUAGE SQL STABLE
+as $f$
+   SELECT DISTINCT ON (a) * FROM test WHERE b >= _ts AND b <= _ts + 2
+$f$;
+
+-- plans must be the same in both cases
+-- specifically, the first plan should not contain the parent hypertable
+-- as that is a sign the pruning was not done successfully
+:PREFIX SELECT * FROM test_f(5);
+
+:PREFIX SELECT DISTINCT ON (a) * FROM test WHERE b >= 5 AND b <= 5 + 2;
```

Comment:
There are a few negative tests missing here: cases where the pruning should *not* be done.

* This was discussed briefly above, but neither of these statements trigger a situation where `resultRelation == 0` and `commandType != CMD_SELECT` so if that is a situation that can occur, it would be good to have a test that you do not mark it for expansion.

* Since CTEs should not be expanded, please add that as a test case to make sure that it works.

* Similarly for queries with row marks (`SELECT ... FOR UPDATE` and friends). Please add tests to make sure that they are not expanded even if they should without the row marks.

Reply:
Would you like to have these tests explicitly in this file? These code paths are actually triggered by existing tests in TimescaleDb (see my comment above for one example).

Reply:
Perhaps to elaborate a bit more - the code path added allows to do pruning of inlineable functions, but the logic `is this a query where a function has been inlined` is not used anywhere in the code. Rather, by just postponing the marking of which tables need pruning to a later stage, when inlining has taken place already, we can just do our pruning as usual. This is why existing *negative* tests actually touch all of these code paths already. They haven't been selected for pruning during the first try in the planner preprocess hook, so they'll get another try in this hook. Whether or not there's an inlineable function doesn't change anything for the negative tests.